Dockerfile.debug

```diff
@@ -0,0 +1,37 @@
+FROM --platform=${BUILDPLATFORM} mcr.microsoft.com/oss/go/microsoft/golang:1.22@sha256:9e5243001d0f43d6e1c556a0ce8eedc1bc80eb37b5210036b289d2c601bc08b6 AS go
```

Comment:
Can we unify the dockerfiles and pass the debug in as a build-arg?
`if [ "${DEBUG}" = "1" ]; then
    extra_flags=...
fi
go build ${extra_fiags} ...
```